ui/src/core_components/embeds/CoreGoogleMaps.vue

```diff
@@ -0,0 +1,223 @@
+import { Loader } from "@googlemaps/js-api-loader"
+
+const rootEl: Ref<HTMLElement> = ref(null);
+const mapEl: Ref<HTMLElement> = ref(null);
+const fields = inject(injectionKeys.evaluatedFields);
+let map : google.maps.Map | null = null;
+let markers = [];
+
+const center = computed<{ lat: number; lng: number; }>(() => ({ lat: fields.lat.value, lng: fields.lng.value }));;
+const mapType = computed(() => (
+	['roadmap', 'satellite', 'hybrid', 'terrain'].includes(fields.mapType.value) 
+		? fields.mapType.value 
+		: 'roadmap'
+));
+
+
+const initMap = async () => {
+	clearMarkers();
+	const loader = new Loader({
+		apiKey: fields.apiKey.value,
+		version: "weekly",
+	});
+	await loader.load();
+	map = new google.maps.Map(mapEl.value, {
+		center: center.value,
+		mapTypeId: mapType.value,
+		zoom: 8,
+		mapId: fields.mapId.value,
+	});
+
+	map.addListener('click', (e: any) => {
+		const event = new CustomEvent("gmap-click", {
+			detail: {
+				payload: e.latLng.toJSON(),
+			},
+		});
+		rootEl.value.dispatchEvent(event);
+	});
+
+	buildMarkers();
+}
+
+const buildMarkers = async () => {
+	if (!fields.mapId.value) return;
+	if (!fields.markers.value) return;
+	const { AdvancedMarkerElement } = await google.maps.importLibrary("marker") as google.maps.MarkerLibrary;
+	clearMarkers();
+	try {
+		const markersData = fields.markers.value as ({ lat: number; lng: number; name: string; }[])
+		markersData.forEach((markerData) => {
+			const marker = new AdvancedMarkerElement({
+				position: {
+					lat: markerData.lat,
+					lng: markerData.lng,
+				},
+				map,
+				title: markerData.name,
+			});
+
+			markers.push(marker);
+
+			marker.addListener('click', () => {
+				const event = new CustomEvent("gmap-marker-click", {
+					detail: {
+						payload: markerData,
+					},
+				});
+				rootEl.value.dispatchEvent(event);
+			});
+		});
+	} catch (e) {
+		console.log(e);
+	}
+};
+
+const clearMarkers = () => {
+	markers.forEach((marker) => marker.setMap(null));
+	markers = [];
+};
+
+onMounted(async () => {
+	if (fields.apiKey.value) {
+		await initMap();
+	}
+});
+watch(center, async (newVal) => {
+	if (map) {
+		map.setCenter(newVal);
+	}
+});
+watch(fields.zoom, async (newVal) => {
+	if (map) {
+		map.setZoom(newVal);
+	}
+});
+watch(mapType, async (newVal) => {
+	if (map) {
+		map.setMapTypeId(newVal);
+	}
+});
+
+watch(fields.markers, buildMarkers);
+watch(fields.apiKey, initMap);
+watch(fields.mapId, initMap);
+
+
+</script>
+
+<style scoped>
+@import "../../renderer/sharedStyles.css";
+
+.CoreGoogleMaps {
+	position: relative;
+	width: 100%;
+	height: 80vh;
+}
+
+.CoreGoogleMaps .map{
+  width: 100%;
+  height: 100%;
+  display: block;
+  margin: auto;
+  border: 1px solid var(--separatorColor);
+}
+
+.CoreGoogleMaps .mask {
+  pointer-events: none;
+}
+
+.CoreGoogleMaps.beingEdited .mask {
+  pointer-events: all;
+  position: absolute;
+  top: 0;
+  left: 0;
+  width: 100%;
+  height: 100%;
+  background-color: rgba(0, 0, 0, 0);
```

Comment:
Any reason for this transparent background-color? Just curious

Reply:
removed